packages/cli/generation/ir-generator/src/resolvers/EndpointResolver.ts

```diff
@@ -17,21 +25,79 @@ interface RawEndpointInfo {
 export class EndpointResolverImpl implements EndpointResolver {
     constructor(private readonly workspace: FernWorkspace) {}
 
-    public resolveEndpointOrThrow({ endpoint, file }: { endpoint: string; file: FernFileContext }): ResolvedEndpoint {
-        const resolvedEndpoint = this.resolveEndpoint({ endpoint, file });
+    public async resolveEndpointOrThrow({
+        endpoint,
+        file
+    }: {
+        endpoint: string;
+        file: FernFileContext;
+    }): Promise<ResolvedEndpoint> {
+        const resolvedEndpoint = await this.resolveEndpoint({ endpoint, file });
         if (resolvedEndpoint == null) {
             throw new Error("Cannot resolve endpoint: " + endpoint + " in file " + file.relativeFilepath);
         }
         return resolvedEndpoint;
     }
 
-    public resolveEndpoint({
+    public async resolveEndpointByMethodAndPath({
```

Comment:
Not necessarily actionable here (can do this in a follow-up if you'd prefer), but we should reuse this when we convert the ReadmeConfig. 

Previously we made a [ReadmeEndpointCache](https://github.com/fern-api/fern/blob/af5350c74836fc2310121ad66b649e8d66555fb5/packages/cli/generation/ir-generator/src/converters/convertReadmeConfig.ts#L40) type since this format was specific to `README.md` generation, but we should be able to use the `EndpointResolver` instead now.

Reply:
The readme code path wasn't entirely clear to me, will do a separate PR for this